src/java/htsjdk/samtools/cram/structure/ReadTag.java

```diff
@@ -374,10 +377,12 @@ public static Object readSingleValue(final byte tagType,
             case 'I':
                 final long val = byteBuffer.getInt() & 0xffffffffL;
                 if (val <= Integer.MAX_VALUE) {
-                    return (int) val;
+                    return (int)val;
                 }
-                throw new RuntimeException(
-                        "Tag value is too large to store as signed integer.");
+                SAMUtils.processValidationError(new SAMValidationError(SAMValidationError.Type.TAG_VALUE_TOO_LARGE,
+                        "Tag value " + val + " too large to store as signed integer.", null), validationStringency);
+                // convert to unsigned int stored in a long
+                return val;
```

Comment:
I don't think this is the behavior we want here... When writing a bam, if the attribute value is a `Long` that is > `BinaryTagCodec.MAX_INT` but <= `BinaryTagCodec.MAX_UINT`, we assign it a tag type of "I" (as opposed to "i"). This is consistent with the BAM spec. So when reading attributes back in, if the attribute type is "I" (rather than "i") we should allow all values >= 0 and <= `BinaryTagCodec.MAX_UINT` rather than raising a validation error if the value is too large to be stored in a signed int.